### PR TITLE
AKS Service Target: Remove automatic secret creation

### PR DESCRIPTION
**Pull request by wbreza**
## Background

Our initial AKS implementation created an `azd` resource in the target k8s namespace with all the azd environment key/value pairs.  These values could then be easily referenced by other resources.

With the change in templating in #2455 the new recommended approach would be for apps to create their own config maps as needed and reference azd secrets via templating.

This brings azd to a more cloud native state where we aren't creating any resources that haven't been explicitly defined.  If users still wanted a config map that contains all azd resources the templating system allows ranging over maps which can be used to achieve the same goal.

## Breaking Change

If users were relying on this `automatic` secret generation then this would be considered a breaking change for them and users will now need to manually create any config maps that their apps/services need.

## Usage
**Create `config.tmpl.yaml`**
```yaml
# config.tmpl.yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: todo-api-config
data:
  AZURE_AKS_IDENTITY_CLIENT_ID: {{.Env.AZURE_AKS_IDENTITY_CLIENT_ID}}
  AZURE_KEY_VAULT_ENDPOINT: {{.Env.AZURE_KEY_VAULT_ENDPOINT}}
  APPLICATIONINSIGHTS_CONNECTION_STRING: {{.Env.APPLICATIONINSIGHTS_CONNECTION_STRING}}
```
**Reference config map from deployment or other k8s resource**
```yaml
# deployment.tmpl.yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: todo-api
spec:
  replicas: 2
  selector:
    matchLabels:
      app: todo-api
  template:
    metadata:
      labels:
        app: todo-api
    spec:
      containers:
        - name: todo-api
          image: {{ .Env.SERVICE_API_IMAGE_NAME }}
          ports:
            - containerPort: 3100
          env:
            - name: AZURE_CLIENT_ID
              valueFrom:
                configMapKeyRef:
                  name: todo-api-config
                  key: AZURE_AKS_IDENTITY_CLIENT_ID
                  optional: false
            - name: AZURE_KEY_VAULT_ENDPOINT
              valueFrom:
                configMapKeyRef:
                  name: todo-api-config
                  key: AZURE_KEY_VAULT_ENDPOINT
                  optional: false
            - name: APPLICATIONINSIGHTS_CONNECTION_STRING
              valueFrom:
                configMapKeyRef:
                  name: todo-api-config
                  key: APPLICATIONINSIGHTS_CONNECTION_STRING
                  optional: false

```

### CODE REVIEW

**Comment by wbreza** [comment]
> Is there an issue associated to this PR? I'd like to understand why do we want to do this?, like, is there some security concern with the current approach or something which is making us introduce the break here?

No issue logged but we've heard direct feedback from the AKS team and CSA's that this is odd behavior.  This was more of a stop-gap workaround solution needed at the time to expose some azd environment variables to the AKS cluster.

**Comment by carmodyquinnms** [comment]
@wbreza 
This looks good.  Yes the creation of kubernetes secrets was unexpected behavior.  The creation of variables in the .env file was the expected behavior.  I think this is a good solution.  The only thing I am wondering about is the format of the variables you show in the configmap is {{ Env.SERVICE_API_IMAGE_NAME }} and I thought the variable replacement format was ${SERVICE_API_IMAGE_NAME}

**Comment by wbreza** [comment]
> @wbreza This looks good. Yes the creation of kubernetes secrets was unexpected behavior. The creation of variables in the .env file was the expected behavior. I think this is a good solution. The only thing I am wondering about is the format of the variables you show in the configmap is {{ Env.SERVICE_API_IMAGE_NAME }} and I thought the variable replacement format was ${SERVICE_API_IMAGE_NAME}

Thanks for the feedback.  We are proposing that we change the templating language used here.  You can read more about that choice in #2455.

**TLDR**: Syntax is the same as used in Helm which puts the user on a good path if/when they choose to adopt a more robust templating system.

**Review by ellismg — COMMENTED** [review]
Happy with the change here, and I think this is the right move.  I do want to spend a few moments with @savannahostrowski when she gets back to have us talk about how we want to roll out this change (as well as #2455) due to breaking change nature of the thing.  Just want to make sure we all have a credible plan for how we roll out these AZD changes + the required changes to our templates so users don't get caught off guard.

So holding off on the green check mark for now, but once we have an answer there, I will approve this.  I love the diff, so much red :-)!

**Review by ellismg — APPROVED** [review]
LGTM - As part of the release this lands in, let's make sure we produce some sort of collateral (doesn't need to be fancy) to show folks with existing deployments how to adopt these changes.

Really love having less magic in the model here.  Great Change!
